src/Codegen/RefCache.php

```diff
@@ -5,8 +5,13 @@
 use namespace HH\Lib\C;
 
 final class RefCache {
+  const type TCachedRef = shape(
```

Comment:
To save memory I stopped storying the whole `Codegen` instance here. This should be everything we need.